src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -768,9 +770,11 @@ def _extract_loc_set_col_info(
             union_data_columns
         ):
             raise ValueError(CANNOT_REINDEX_ON_DUPLICATE_ERROR_MESSAGE)
+        # split labels into existing and new
         new_column_pandas_labels = [
             label for label in columns if label not in frame_data_columns
         ]
+        columns = [label for label in columns if label in frame_data_columns]
```

Comment:
nit - I would rename `columns` here to `existing_column_pandas_labels` to match with the comment `# split labels into existing and new` and the other variable `new_column_pandas_labels`.

Reply:
We cannot rename this because the variable is used elsewhere in the code. This is needed because we throw `KeyError` now